src/python/grpcio_tests/tests/unit/_server_ssl_cert_config_test.py

```diff
@@ -263,38 +266,32 @@ def _do_common_test_recipe(self):
 
 class ServerSSLCertConfigFetcherParamsChecks(unittest.TestCase):
 
-    def test_params_checks(self):
-        for initial_config, subsequent_config_fetcher, exc_msg in (
-            (None, str, 'initial certificate config must be specified'),
-            (1, '1', 'certificate config fetcher must be callable'),):
-            with self.assertRaises(ValueError) as exc:
-                grpc.ssl_server_credentials_dynamic_cert_config(
-                    initial_config, subsequent_config_fetcher)
-            self.assertEqual(exc.exception.message, exc_msg)
+    def test_check_on_initial_config(self):
+        with self.assertRaises(ValueError):
+            grpc.ssl_server_credentials_dynamic_cert_config(None, str)
 
+    def test_check_on_config_fetcher(self):
+        with self.assertRaises(ValueError):
+            grpc.ssl_server_credentials_dynamic_cert_config(1, '1')
 
-class ServerSSLCertReloadTestWithClientAuth(_ServerSSLCertReloadTest,
-                                            unittest.TestCase):
 
-    def setUp(self):
-        self._do_setUp(True)
+class ServerSSLCertReloadTestWithClientAuth(_ServerSSLCertReloadTest):
 
     def test(self):
         self._do_common_test_recipe()
 
 
-class ServerSSLCertReloadTestWithoutClientAuth(_ServerSSLCertReloadTest,
-                                               unittest.TestCase):
+class ServerSSLCertReloadTestWithoutClientAuth(_ServerSSLCertReloadTest):
 
-    def setUp(self):
-        self._do_setUp(False)
+    def __init__(self, *args, **kwargs):
+        super(ServerSSLCertReloadTestWithoutClientAuth, self).__init__(*args, **kwargs)
+        self.with_client_auth = False
```

Comment:
i'm not super stoked about having to toggle the test `with_client_auth` this way, and about the use of `__init__()`, as it appears (per stackoverflow :)) that very rarely does one need to do that for unit test cases.

Reply:
... so why not make `with_client_auth` a parameter of `do_common_test_recipe`?

Reply:
because the test needs `with_client_auth` at `setUp()` time in order to initialize the server credentials, to start the server.

Reply:
Let's not overthink this. This is an abstract super class. How do abstract super classes ask for things from concrete subclasses? They define abstract methods and they call those abstract methods and subclasses implement those abstract methods.

So define a
```py
@abc.abstractmethod
def with_client_auth(self):
  raise NotImplementedError()
```
in the superclass and implement it in the subclasses?

Reply:
i've gone with
```
def require_client_auth(self):
  return True
```
in the base class

Reply:
Please don't? I don't think there's a single occurrence of overriding a concrete method implementation in gRPC Python (outside of `setUp` and `tearDown`?); let's not start now.

Reply:
sounds good! fixed